src/Features/Core/Portable/EditAndContinue/EditAndContinueDiagnosticUpdateSource.cs

```diff
@@ -104,7 +104,7 @@ public void ReportDiagnostics(Workspace workspace, Solution solution, ImmutableA
 
             if (documentDiagnostics.Length > 0)
             {
-                foreach (var (documentId, diagnosticData) in documentDiagnostics.ToDictionary(data => data.DocumentId!))
+                foreach (var (documentId, diagnosticData) in documentDiagnostics.ToDictionary(static data => data.DocumentId!))
```

Comment:
>ToDictionary [](http://example.com/codeflow?start=81&length=12)

Why even ToDictionary here?

Reply:
afaict, it's so that if there's multiple diagnostics with teh same ID that they get grouped together, and the group is reported at once.  Why that's the pattern here is not something i'm particularly knowledgeable about as i don't know much about EnC.

Reply:
Gotcha, I needed to look closer into what ToDictionary was doing